### DISCUSSION TITLE
How to grow popover content width to match trigger

### DISCUSSION THREAD

**Discussion by StephenSHorton** [radix-ui/primitives#2894] [Help] [answered]
I have a similar question to #1921 but my popover content is too small and doesn't grow to the size of the trigger?
![Screenshot 2024-05-10 at 10 35 46 AM](https://github.com/radix-ui/primitives/assets/54323940/7beb1733-4211-45c6-a2f2-935665ed4e47)
I've got `w-full` or `width: 100%` on both my trigger and the content. Is it not possible to do this? Do I have to have a fixed width?

**Comment by zahidiqbalnbs** [comment, marked as answer]
Check here. https://www.radix-ui.com/primitives/docs/components/popover#constrain-the-content-size

**Reply by StephenSHorton** [reply]
~~Thanks, but I'm looking for a dynamic solution that doesn't rely on a set width.~~

**Reply by ajayvignesh01** [reply]
This is a dynamic solution tho.

Just add this to your content:
```tsx
style={{ width: 'var(--radix-popover-trigger-width)' }}
```

**Reply by StephenSHorton** [reply]
I stand corrected, thank you!

**Reply by ali-sahi** [reply]
Thankyou @ajayvignesh01 i have been wrapping my head around with this for a while now and your solution worked. 
By the way i still don't know why width:100% is not working. Its working if we define it in pixels. 
Do you have any idea?

**Reply by LukasKri** [reply]
I didn't want to use `style`, so this worked:
```
<PopoverContent
      className='w-[--radix-popover-trigger-width]'
>
```

**Reply by rihid** [reply]
working good, upvoted!!

**Reply by Hadeed-S** [reply]
for cases where we have a trigger using input using slots for icons / buttons

`<TextField.Root
        size={size}
        color={color}
        variant={variant}
        placeholder={placeholder}
        className={className}
        {...props}
      >
        <TextField.Slot side={'left'}>{prefix}</TextField.Slot>
        <TextField.Slot side={'right'}>{suffix}</TextField.Slot>
      </TextField.Root>`
      
the content does not span the entire trigger although the input encapsulates the slot, Help would be appreciated

**Reply by Kudze** [reply]
@LukasKri shouldn't it be `w-[var(--radix-popover-trigger-width)]`?

**Reply by mustafazeydani** [reply]
> @LukasKri shouldn't it be `w-[var(--radix-popover-trigger-width)]`?

w-(--radix-popover-trigger-width) for tailwind v4

**Reply by ajayvignesh01** [reply]
> @LukasKri shouldn't it be `w-[var(--radix-popover-trigger-width)]`?

If you're using tailwind v3, yes

**Comment by Ryanjso** [comment]
I am facing the same issue, were you able to identify a solution?

**Reply by StephenSHorton** [reply]
Yes, the accepted answer is the solution. You need to leverage the `--radix-popover-trigger-width` css variable. (See [here](https://www.radix-ui.com/primitives/docs/components/popover#constrain-the-content-size))

**Comment by BartekCK** [comment]
If you are using Shadcn and Tailwind v4, you can do something like this:
```tsx
<PopoverContent className="p-0 w-[var(--radix-popover-trigger-width)]"> {...} </PopoverContent>
```

**Reply by rgomezv** [reply]
This worked for me too in Tailwind V4 with Shadcn.

**Reply by gijzy** [reply]
Thank you

**Reply by gdgenchev** [reply]
sick bro, thanks!

**Reply by mk-logbuy** [reply]
legend
